assets/predictive-search.js

```diff
@@ -27,7 +27,16 @@ class PredictiveSearch extends SearchForm {
 
   onChange() {
     super.onChange();
-    this.searchTerm = this.getQuery();
+    const newSearchTerm = this.getQuery();
+    if (!this.searchTerm || !newSearchTerm.startsWith(this.searchTerm)) {
+      // Hide the results when they are no longer relevant for the new search term
+      this.closeResults(false);
```

Comment:
This does what the function you had before did 🤔 
Here are two video comparisons: [using clearResources()](https://screenshot.click/26-37-z1bah-zoj8s.mp4) & [using closeResults(false)](https://screenshot.click/26-41-laoke-eu6yn.mp4)

Reply:
I tested and it seemed to be fine, but maybe my internet was too fast to appreciate the difference. Thanks for the thorough testing!